src/AggregateFunctions/UniqExactSet.h

```diff
@@ -127,6 +128,10 @@ class UniqExactSet
         else
         {
             auto & lhs = asTwoLevel();
+
+            if (other.isSingleLevel())
```

Comment:
it appears to be beneficial to consider this case, otherwise we will do implicit conversion of `other` to two-level on the next line (`other.getTwoLevelSet()`). it is not the end of the world, but noticeable.